lib/ruleProgress.js

```diff
@@ -0,0 +1,62 @@
+/* global -Promise */
+'use strict';
+
+var _ = require('underscore'),
+    hmdaRuleSpec = require('hmda-rule-spec');
+
+var RuleProgress = (function() {
+    return function() {
+
+        /**
+         * calculates the estimate for the number of parsedRules that will need to be executed 
+         * @param  {year}  which year to use for the rule spec
+         * @param  {scopeList}  an array containing the scoped specs to look up for this edit type
+         * @param  {type}  the type of HMDA edits (syntactical, validity, etc.)
+         */
+        this.calcEstimatedTasks = function (year, scopeList, type) {
+            switch (type) {
```

Comment:
Why does this need a switch statement? Shouldn't this just be an `if (type) {...}`?